samples/matter/common/src/diagnostic/diagnostic_logs_provider.cpp

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifdef CONFIG_NCS_SAMPLE_MATTER_DIAGNOSTIC_LOGS
+
+#include <app/util/config.h>
+
+#ifndef EMBER_AF_DIAGNOSTIC_LOGS_CLUSTER_SERVER_ENDPOINT_COUNT
+#error "Diagnostics cluster is not enabled in the root endpoint"
+#endif
+
+#include "diagnostic_logs_provider.h"
+
+#include <app/clusters/diagnostic-logs-server/diagnostic-logs-server.h>
+
+using namespace chip;
+using namespace chip::app::Clusters::DiagnosticLogs;
+using namespace Nrf;
+using namespace Nrf::Matter;
+
+DiagnosticLogProvider DiagnosticLogProvider::sInstance;
+
+namespace
+{
+bool IsValidIntent(IntentEnum intent)
+{
+	return intent < IntentEnum::kUnknownEnumValue;
+}
+
+const char *getIntentStr(IntentEnum intent)
+{
+	switch (intent) {
+	case IntentEnum::kEndUserSupport:
+		return "end user";
+	case IntentEnum::kNetworkDiag:
+		return "network";
+	case IntentEnum::kCrashLogs:
+		return "crash";
+	default:
+		return "unknown";
+	}
+
+	return "unknown";
+}
+} /* namespace */
+
+CHIP_ERROR DiagnosticLogProvider::GetCrashLogs(chip::MutableByteSpan &outBuffer, bool &outIsEndOfLog)
+{
+	return CHIP_ERROR_NOT_IMPLEMENTED;
+}
+
+CHIP_ERROR DiagnosticLogProvider::ClearCrashLogs()
+{
+	return CHIP_ERROR_NOT_IMPLEMENTED;
+}
+
+CHIP_ERROR DiagnosticLogProvider::StartLogCollection(IntentEnum intent, LogSessionHandle &outHandle,
+						     Optional<uint64_t> &outTimeStamp,
+						     Optional<uint64_t> &outTimeSinceBoot)
+{
+	VerifyOrReturnError(IsValidIntent(intent), CHIP_ERROR_INVALID_ARGUMENT);
+
+	uint16_t freeSlot = mIntentMap.GetFirstFreeSlot();
+	VerifyOrReturnError(freeSlot != kInvalidLogSessionHandle, CHIP_ERROR_NO_MEMORY);
```

Comment:
I think that `freeSlot` will never be equal to kInvalidLogSessionHandle (UINT16_MAX) unless you set the size of the map to UINT16_MAX-1. 

The invalid slot number is represented as the map size + 1, therefore in our case the invalid slot will be NCS_SAMPLE_MATTER_DIAGNOSTIC_LOGS_MAX_SIMULTANEOUS_SESSIONS + 1 (21 by default).